File: leetcode/ag_384.py
import random
import copy

class Solution(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        :type size: int
        """
        self.nums = copy.deepcopy(nums)
        self.orig_nums = copy.deepcopy(nums)
        

    def reset(self):
        """
        Resets the array to its original configuration and return it.
        :rtype: List[int]
        """
        self.nums = copy.deepcopy(self.orig_nums)
        return self.nums
        
    def show(self):
        print self.nums

    def shuffle(self):
        """
        Returns a random shuffling of the array.
        :rtype: List[int]
        """
        random.shuffle(self.nums)
        return self.nums


def run():
    nums =[1,2,3,4,9,12,24]
    ns = Solution(nums)
    print ns.shuffle()
    print ns.reset()
    print ns.shuffle()
    print ns.reset()

File: leetcode/ag_384.cpp
#include <iostream>
#include <vector>


using namespace std;

class Solution {
private:
    std::vector<int> nums;
    std::vector<int> origin_nums;

public:
    Solution(vector<int> nums) {
        this->nums = nums;
        this->origin_nums = nums;
    }
    
    /** Resets the array to its original configuration and return it. */
    vector<int> reset() {
        this->nums = this->origin_nums;
        return this->nums;
    }
    
    /** Returns a random shuffling of the array. */
    vector<int> shuffle() {
        auto len = this->nums.size();
        for(auto i = 0 ; i< len; i++){
            swap(this->nums[i] , this->nums[rand() % len]);
        }
        return this->nums;
    }
    void print()
    {
        for(auto i : this->nums){
            std::cout<<i<<"  ";
        }
        std::cout<<std::endl;
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * vector<int> param_1 = obj.reset();
 * vector<int> param_2 = obj.shuffle();
 */

int main()
{

    std::vector<int> ww = {1,2,3,4,5};
    Solution *s = new Solution(ww);
    s->shuffle();
    s->print();

}
